### PR TITLE
[nghttp3] Update to v1.5.0

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@talregev, Thanks for your PR, I tested the usage by `nghttp3:x64-windows`, I found that the project could not reference the header files under `${VCPKG_ROOT_DIR}\installed\${TRIPLET}\include`.

**Comment by talregev** [comment]
@JonLiu1993 Thank you for your feedback! 
Do you have suggestion how to solve it? 
@dg0yt Do you have any comments?

**Comment by talregev** [comment]
@JonLiu1993 
I notice that nghttp3 already export the cmake. I change the location. 
I am not sure it solve the problem that you face.
Do you mind to check it again?

**Comment by talregev** [comment]
@JonLiu1993 
I did another update after my last comment.

**Review by dg0yt — COMMENTED** [review]
Please drop the patch.
The manifest needs a host dependency for `vcpkg-cmake-config`.
The portfile needs to call `vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/nghttp3")`.
(Untested, just studying the PR.)

**Comment by talregev** [comment]
> Please drop the patch. The manifest needs a host dependency for `vcpkg-cmake-config`. The portfile needs to call `vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/nghttp3")`. (Untested, just studying the PR.)

Done.
Thank you for your help.

**Comment by talregev** [comment]
@JonLiu1993 Please review again.
Thank you for your hard work!

**Review by dg0yt — APPROVED** [review]
LGTM but someone else must approve.

File: ports/nghttp3/portfile.cmake [hunk 1]
```diff
@@ -2,15 +2,16 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ngtcp2/nghttp3
     REF v${VERSION}
-    SHA512 56b5ac28f01a9210f0bd3de06dabe9f34867b72c68df0776f8faf3f03a4b82185fd16891f67f710b8c7db56523e29a7c51e4b2337ea9e5331bc6c21a26ed12dd
+    SHA512 d8b6db7a1323e036cd2d1aab1ded299e83024ce451cddbfe0ea102d968bdcb57221dbcc231b73880e0987cf3ed7ecd2c2b5f53b10947d9accb7603d7c3fcbb95
     HEAD_REF main
+    PATCHES
```
Comment (JonLiu1993) [inline ports/nghttp3/portfile.cmake:7]:
```suggestion
```

**Comment by talregev** [comment]
@JonLiu1993 Can you also test usage?

File: ports/nghttp3/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/nghttp3/portfile.cmake]: Done!

**Comment by talregev** [comment]
@MonicaLiu0311 Can you help for review the PR?
Thank you for your help 🙏🏻

**Comment by JonLiu1993** [comment]
This patch can fix usage issue.
[fix-include-usage.patch](https://github.com/user-attachments/files/16793004/fix-include-usage.patch)

**Comment by dg0yt** [comment]
Reduce the extra lines to:
~~~
	target_include_directories(nghttp3 INTERFACE $<INSTALL_INTERFACE:include>)
~~~
Everything else is covered.

And this should be upstreamed.

**Comment by talregev** [comment]
@JonLiu1993 Thank you for your help 🙏🏻🙏🏻 
I will add this patch!.

**Comment by talregev** [comment]
> Reduce the extra lines to:
> 
> ```
> 	target_include_directories(nghttp3 INTERFACE $<INSTALL_INTERFACE:include>)
> ```
> 
> Everything else is covered.
> 
> And this should be upstreamed.

@dg0yt This is needed for both static and share? or only share as @JonLiu1993 suggested?

**Comment by talregev** [comment]
I added both static and shared.

**Comment by talregev** [comment]
@JonLiu1993 Can you try both for windows and linux.
Linux:
```
nghttp3 provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(nghttp3 CONFIG REQUIRED)
  target_link_libraries(main PRIVATE nghttp3::nghttp3_static)
```

Windows:
```
nghttp3 provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(nghttp3 CONFIG REQUIRED)
  target_link_libraries(main PRIVATE nghttp3::nghttp3)
```

**Comment by dg0yt** [comment]
> This is needed for both static and share?

Yes. It is needed for any direct usage of the lib, at compile time. 

(Static libs have transitive usage requirements at link time. But transitive requirements can also exist for other contexts.)

> ~~~
> # this is heuristically generated, and may not be correct
>~~~

Note that the difference is related to linkage, not system. Do give a uniform instruction, there should be a `usage` file:
~~~cmake
nghttp3 provides CMake targets:

  find_package(nghttp3 CONFIG REQUIRED)
  target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:nghttp3::nghttp3>,nghttp3::nghttp3,nghttp3::nghttp3_static>)
~~~

**Comment by talregev** [comment]
> Note that the difference is related to linkage, not system

I know.

**Comment by talregev** [comment]
Done.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Tested usage successfully by `nghttp3:x64-windows`:
````
nghttp3 provides CMake targets:

  find_package(nghttp3 CONFIG REQUIRED)
  target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:nghttp3::nghttp3>,nghttp3::nghttp3,nghttp3::nghttp3_static>)
````

**Pull request merged**